huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go

```diff
@@ -267,13 +294,91 @@ func resourceDmsRabbitmqInstanceCreate(ctx context.Context, d *schema.ResourceDa
 		}
 	}
 
+	createOpts := &instances.CreateOps{
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		Engine:              engineRabbitMQ,
+		EngineVersion:       d.Get("engine_version").(string),
+		StorageSpace:        d.Get("storage_space").(int),
+		AccessUser:          d.Get("access_user").(string),
+		VPCID:               d.Get("vpc_id").(string),
+		SecurityGroupID:     d.Get("security_group_id").(string),
+		SubnetID:            d.Get("network_id").(string),
+		AvailableZones:      availableZones,
+		ProductID:           d.Get("flavor_id").(string),
+		BrokerNum:           d.Get("broker_num").(int),
+		MaintainBegin:       d.Get("maintain_begin").(string),
+		MaintainEnd:         d.Get("maintain_end").(string),
+		SslEnable:           d.Get("ssl_enable").(bool),
+		StorageSpecCode:     d.Get("storage_spec_code").(string),
+		EnterpriseProjectID: common.GetEnterpriseProjectID(d, cfg),
+	}
+
+	if pubIpID, ok := d.GetOk("public_ip_id"); ok {
+		createOpts.EnablePublicIP = true
+		createOpts.PublicIpID = pubIpID.(string)
+	}
+
+	// set tags
+	if tagRaw := d.Get("tags").(map[string]interface{}); len(tagRaw) > 0 {
+		createOpts.Tags = utils.ExpandResourceTags(tagRaw)
+	}
+
+	log.Printf("[DEBUG] Create DMS RabbitMQ instance Options: %#v", createOpts)
+	// Add password here so it wouldn't go in the above log entry
+	createOpts.Password = d.Get("password").(string)
+
+	v, err := instances.CreateWithEngine(client, createOpts).Extract()
+	if err != nil {
+		return diag.Errorf("error creating DMS RabbitMQ instance: %s", err)
+	}
+	instanceID := v.InstanceID
+	// Store the instance ID now
+	d.SetId(instanceID)
+	log.Printf("[INFO] Creating RabbitMQ instance, ID: %s", instanceID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"CREATING"},
+		Target:       []string{"RUNNING"},
+		Refresh:      rabbitmqInstanceStateRefreshFunc(client, instanceID),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        300 * time.Second,
+		PollInterval: 15 * time.Second,
+	}
+
+	if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+		return diag.Errorf("error waiting for RabbitMQ instance (%s) to be ready: %s", instanceID, err)
+	}
+
+	return nil
+}
+
+func createRabbitMQInstanceWithProductID(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.DmsV2Client(region)
+	if err != nil {
+		return diag.Errorf("error initializing DMS RabbitMQ(v2) client: %s", err)
+	}
+	var availableZones []string // Available zones IDs
+	azIDs, ok := d.GetOk("available_zones")
```

Comment:
the `availability_zones` is the recommand parameter, not `available_zones`.

Reply:
the param of available_zones is used in previous versions, so it is needed to be compatible